test/services/SubmissionServiceSpec.scala

```diff
@@ -41,156 +43,286 @@ class SubmissionServiceSpec
     with MockSoleTraderMatchingService
     with MockBusinessVerificationService
     with MockCreateTrnService
+    with MockRegistrationOrchestrationService
     with FeatureSwitching {
 
   object TestService extends SubmissionService(
-    mockJourneyService,
     mockSoleTraderMatchingService,
     mockSoleTraderIdentificationService,
     mockBusinessVerificationService,
-    mockCreateTrnService)
+    mockCreateTrnService,
+    mockRegistrationOrchestrationService
+  )
 
   implicit val hc: HeaderCarrier = HeaderCarrier()
 
+  "submit: sole trader journey with businessVerificationCheck = false" when {
+
+      "the user has a sautr" should {
+        s"register Without Business Verification" in {
+
+          val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+          mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetails)))
+          mockMatchSoleTraderDetails(testJourneyId, testIndividualDetails, journeyConfigWithoutBV)(Future.successful(Right(true)))
+          mockStoreIdentifiersMatch(testJourneyId, identifiersMatch = true)(Future.successful(SuccessfullyStored))
+          mockRegisterWithoutBusinessVerification(testJourneyId, testIndividualDetails.optNino, testIndividualDetails.optSautr.get)(Future.successful(Registered("someId")))
+
+          val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+          result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl + s"?journeyId=$testJourneyId")
+
+        }
+      }
+
+    "the user has no sautr" should {
+      s"not be registered"  in {
+        val matchingResult = Random.nextBoolean()
+
+        val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+        mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetailsNoSautr)))
+        mockMatchSoleTraderDetails(testJourneyId, testIndividualDetailsNoSautr, journeyConfigWithoutBV)(Future.successful(Right(matchingResult)))
+        mockStoreIdentifiersMatch(testJourneyId, identifiersMatch = true)(Future.successful(SuccessfullyStored))
+        mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
+
+        val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+        result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl)
+
+        verifyStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)
```

Comment:
Here we could also verify that the IdentifiersMatch has been stored correctly. Also throughout this whole file the verifies are not very consistent, they could be being used in the whole file I think

Reply:
I agree they should use everywhere ... in the new code for the moment ;-)

Reply:
....  Milly I dont think we are using storeIdentifiersMatch in the code at all  ..... I think that line is useless .... because we are mocking things and I think we copied from another test ... where it was useless too. I removed it and no verify is needed